docs/rules/no-array-index-key.md

```diff
@@ -0,0 +1,23 @@
+# Prevent usage of Array index in keys
+
+Warn if an element uses an Array index in its `key`.
```

Comment:
I think adding a short description of why you might want to enable this rule would be good. Something like:

```
It's a bad idea to use the array index since it doesn't uniquely identify your elements. In cases where the array is sorted, the index will be changed, even though the element representing that index may be the same -- resulting in unnecessary render cycles.
```